helix-view/src/register.rs

```diff
@@ -0,0 +1,191 @@
+use crate::info::Info;
+use std::{collections::HashMap, fmt::Display, str::FromStr};
+use strum_macros::{AsRefStr, EnumString};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EnumString, AsRefStr)]
+pub enum Register {
```

Comment:
Let's keep a `char` as the register identifier, since that was the intent of the original implementation. There is no special "yank" register, just a default slot that's usually used for yanking but can be overriden. In fact all these extra dependencies would be shaken out if we just used a `const Register::YANK = "\""` constant